geomagio/JSON/JSONWriter.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import absolute_import
+from builtins import range
+
+from io import BytesIO
+from collections import OrderedDict
+from datetime import datetime
+import json
+import numpy
+import textwrap
+from .. import ChannelConverter, TimeseriesUtility
+from ..edge import EdgeFactory
+from ..TimeseriesFactoryException import TimeseriesFactoryException
+from ..Util import create_empty_trace
+
+
+class JSONWriter(object):
+    """JSON writer.
+    """
+
+    def __init__(self):
+        self.dictionary = OrderedDict()
+
+    def write(self, out, timeseries, channels, **kwargs):
+        """write timeseries to json file
+
+        Parameters
+        ----------
+        out: file object
+            file object to be written to. could be stdout
+        timeseries: obspy.core.stream
+            timeseries object with data to be written
+        channels: array_like
+            channels to be written from timeseries object
+        """
+        request = kwargs.get('request')
+        for channel in channels:
+            if timeseries.select(channel=channel).count() == 0:
+                raise TimeseriesFactoryException(
+                    'Missing channel "%s" for output, available channels %s' %
+                    (channel, str(TimeseriesUtility.get_channels(timeseries))))
+        stats = timeseries[0].stats
+        if len(channels) != 4:
+            channels = self._pad_to_four_channels(timeseries, channels)
+        self._format_metadata(stats, channels)
+        if request:
+            self.dictionary['metadata']['url'] = 'http://geomag.usgs.gov/ws/edge/?' + request
+        self._format_times(timeseries, channels)
+        self._format_data(timeseries, channels, stats)
+        out.write(json.dumps(self.dictionary, ensure_ascii=True).encode(
+                'utf8'))
+
+    def _format_metadata(self, stats, channels):
+        """format metadata for json file and update dictionary
+
+        Parameters
+        ----------
+        stats: obspy.core.trace.stats
+            holds the observatory metadata
+        channels: array_like
+            channels to be reported.
+        """
+        dict = self.dictionary.copy()
+        dict['type'] = 'Timeseries'
+        dict['metadata'] = OrderedDict()
+        dict['metadata']['intermagnet'] = OrderedDict()
```

Comment:
Create a local `intermagnet` variable, which can be added the the metadata dictionary once it's complete.  Locals are much more efficient than repeated dictionary lookups